src/workers/filesystems/infrastructure/SyncRemoteFileSystem.ts

```diff
@@ -46,625 +50,747 @@ type CacheData = {
-							concurrency: 10,
-						},
-					}
-				);
-			});
-		},
-
-		async smokeTest() {
-			if (!(await isOnline())) {
-				throw new ProcessFatalError(
-					'NO_INTERNET',
-					createErrorDetails({}, 'Remote smoke test (online test)')
-				);
-			}
-
-			const res = await httpRequest(
-				`${process.env.API_URL}/api/storage/v2/folder/${baseFolderId}`,
-				{ headers }
-			);
-
-			if (!res.ok) {
-				if (res.status === 401) {
-					ipcRenderer.emit('user-is-unauthorized');
-				}
-				throw new ProcessFatalError(
-					'NO_REMOTE_CONNECTION',
-					createErrorDetails(
-						{},
-						'Remote smoke test (get base folder test)',
-						`res: ${await serializeRes(res)}`
-					)
-				);
-			}
-		},
-	};
+  const cache: RemoteCache = {};
+  const createFolderQueue = new EventEmitter().setMaxListeners(0);
+
+  async function getTree(retrieveFrom: 'server' | 'local-db'): Promise<{
+    files: ServerFile[];
+    folders: ServerFolder[];
+  }> {
+    if (retrieveFrom === 'local-db') {
+      Logger.info('Getting tree from local DB');
+      const driveFiles = new DriveFilesCollection();
+      const driveFolders = new DriveFoldersCollection();
+      const remoteSyncManager = new RemoteSyncManager(
+        {
+          files: driveFiles,
+          folders: driveFolders,
+        },
+        {
+          httpClient: getNewTokenClient(),
+          syncFiles: true,
+          syncFolders: true,
+          fetchFilesLimitPerRequest: 50,
+          fetchFoldersLimitPerRequest: 50,
+        }
+      );
+
+      Logger.info('Making sure remote is in sync with local db');
+      await remoteSyncManager.startRemoteSync();
+
+      Logger.info(
+        `Finished remote sync with status ${remoteSyncManager.getSyncStatus()}`
+      );
+      if (!remoteSyncManager.localIsSynced())
+        throw new Error(
+          '[RemoteSyncManager] local is not in sync with remote, tree cannot be generated from local-db'
+        );
+
+      Logger.info('Getting all the files from local DB...');
+      const { result: allDriveFiles } = await driveFiles.getAll();
+      Logger.info('Getting all the folders from local DB...');
+      const { result: allDriveFolders } = await driveFolders.getAll();
+      const serverFiles: ServerFile[] = allDriveFiles.map<ServerFile>(
+        (driveFile) => {
+          const serverFile: ServerFile = {
+            bucket: driveFile.bucket,
+            createdAt: driveFile.createdAt,
+            encrypt_version: '03-aes',
+            fileId: driveFile.fileId,
+            folderId: driveFile.folderId,
+            id: driveFile.id,
+            modificationTime: driveFile.modificationTime,
+            name: driveFile.name,
+            size: driveFile.size,
+            type: driveFile.type,
+            updatedAt: driveFile.updatedAt,
+            userId: driveFile.userId,
+          };
+          return serverFile;
+        }
+      );
+
+      const serverFolders: ServerFolder[] = allDriveFolders.map<ServerFolder>(
```

Comment:
The map can be expressed like: 
allDriveFolders.map<ServerFolder>((folder) => ({bucket: folder.bucket ....})

And there is no need to declare a variable and return it